src/main/java/htsjdk/samtools/util/AbstractLocusIterator.java

```diff
@@ -475,6 +473,19 @@ private void populateCompleteQueue(final Locus stopBeforeLocus) {
         lastPosition = locusInfo.getPosition();
     }
 
+    private void removeSkippedRegion(Locus stopBeforeLocus) {
+        int i = 0;
+        while (i < accumulator.size() && accumulator.get(i).isEmpty() &&
+                locusComparator.compare(accumulator.get(i), stopBeforeLocus) < 0) {
+            i++;
+        }
+        if (i > 0){
+            List<K> temporaryList = new ArrayList<>(accumulator.subList(i, accumulator.size()));
```

Comment:
This seems to do a bunch of unnecessary allocations. 

 I wasn't sure the best way to do this, but according to the java doc for subList

```
  * <p>This method eliminates the need for explicit range operations (of
     * the sort that commonly exist for arrays).  Any operation that expects
     * a list can be used as a range operation by passing a subList view
     * instead of a whole list.  For example, the following idiom
     * removes a range of elements from a list:
     * <pre>
     *      list.subList(from, to).clear();
```

Which will use the efficient internal method `removeRange`.  So, I learned something new.  Would you update to use `subList.clear` instead of new list allocations?

Good to go after that.

Reply:
@lbergelson Thanks for information about this method! I corrected code according to your comment.